src/AppBundle/Controller/DashboardController.php

```diff
@@ -0,0 +1,345 @@
+<?php
+
+namespace AppBundle\Controller;
+
+
+use AppBundle\AppBundle;
+use AppBundle\Form\DashboardType;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Knp\Bundle\PaginatorBundle\KnpPaginatorBundle;
+use AppBundle\Entity\Dashboard;
+use AppBundle\Entity\User;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Ob\HighchartsBundle\Highcharts\Highchart;  //Highcharts bundle
+use Symfony\Component\HttpFoundation\Request;
+use AppBundle\Entity\Component;
+use AppBundle\Form\ComponentType;
+use Symfony\Component\HttpFoundation\Session\Session;
+
+
+
+class DashboardController extends Controller
+{
+
+    public function indexAction()
+    {
+        return $this->render('AppBundle::layout.html.twig'
+        );
+    }
+
+    //.......Dashboard view action
+    /***
+     * @param Request $request
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function viewAction(Request $request, $id)
+    {
+        //.......Get current logged user
+        $user = $this->container->get('security.context')->getToken()->getUser();
+        $userId = $user->getId();
+
+        $repository = $this
+            ->getdoctrine()
+            ->getManager()
+            ->getRepository('AppBundle:Component');
+
+        //.......Get current dashboard components
+        $components = $repository->findBy(
```

Comment:
We cannot access components from dashboard entity because dashboard entity doesn't contain getComponents(), it's  the component entity that has dashboard attribute.